src/main/webapp/components/Auth/Auth.js

```diff
@@ -170,19 +171,18 @@ export default function Auth() {
 	const usernameExists = errors[0];
 	const emailExists = errors[1];
 	if (emailExists || usernameExists) {
-	  setUsernameError(usernameExists);
-	  setEmailError(emailExists);
-	  if (usernameError) {
+	  if (usernameExists) {
+	    setUsernameError(true);	  
             setUsernameErrorMsg('Username is already taken');
 	  }
-	  if (emailError) {
+	  if (emailExists) {
+	    setEmailError(true);
 	    setEmailErrorMsg('Email address has already been used');
 	  }
 	} else {
 	  const signInInfo = { username: username, email: email };
-	  console.log(signInInfo);
 	  //dispatch(signIn(signInInfo));
-	  createHashHistory.push('/editor');
+	  history.push('/editor');
 	}
```

Comment:
There is a lot of nesting here.  It is a lot more readable to early return instead: https://medium.com/swlh/return-early-pattern-3d18a41bba8

I would remove line 173 and do something like:

if (usernameExists) {
  setUsernameError(true);
  setUsernameErrorMsg("msg");
  return;
}
if (emailExists) {
  setEmailError(true);
  setEmailErrorMsg("msg");
  return;
}

Also, we can do something similar - instead of checking if (response.status == 200), we can do

if (response.status != 200) {
  return;
}
// Do everything else here

Reply:
Made changes